scipy/spatial/ckdtree.pyx

```diff
@@ -1035,24 +1060,45 @@ cdef public class cKDTree [object ckdtree, type ckdtree_type]:
         else:
             raise ValueError("Invalid output type") 
 
+    @cython.boundscheck(False)
+    def build_weights(cKDTree self, object weights):
+        cdef np.intp_t num_of_nodes
+        cdef np.ndarray[np.float64_t, ndim=1, mode="c"] node_weights;
+        cdef np.ndarray[np.float64_t, ndim=1, mode="c"] _weights
+
+        num_of_nodes = self.tree_buffer.size();
+        node_weights = np.empty(num_of_nodes, dtype=np.float64)
+        _weights = np.ascontiguousarray(weights, dtype=np.float64)
+
+        if len(_weights) != self.n:
+            raise ValueError('Number of weights differ from the number of data points')
+
+        build_weights(<ckdtree*> self, <np.float64_t*>np.PyArray_DATA(node_weights),
+                            <np.float64_t*> np.PyArray_DATA(weights))
+                    
+        return node_weights
 
     # ---------------
     # count_neighbors
     # ---------------
 
     @cython.boundscheck(False)
-    def count_neighbors(cKDTree self, cKDTree other, object r, np.float64_t p=2.):
+    def count_neighbors(cKDTree self, cKDTree other, object r, np.float64_t p=2., 
+                        object self_weights=None, object other_weights=None, int cumulative=True):
         """
-        count_neighbors(self, other, r, p=2.)
+        count_neighbors(self, other, r, p=2., self_weights=None, other_weights=None, cumulative=True)
```

Comment:
I like clearly readable names, so ``self_weights`` > ``sw``. ``weights`` seems a bit better than ``self_weights`` to me though - if the description in the docstring is clear then it shouldn't be confusing. Another alternative that avoids ``self`` would be ``tree_weights``.